service/controller/resource/clusterdependents/resource.go

```diff
@@ -122,6 +122,11 @@ func (r *Resource) ensureMachinePoolCRsDeleted(ctx context.Context, cr capiv1alp
 	}
 
 	for _, mp := range mpList.Items {
+		if !mp.GetDeletionTimestamp().IsZero() {
```

Comment:
I saw it in CAPI code, I don't think it hurts?

Reply:
Sounds sane to me :)